airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -66,6 +69,46 @@ def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) ->
             params["order"] = f"{self.order_field} asc"
             params[self.filter_field] = self.default_filter_field_value
         return params
+    
+    # To handle certificate error when making requests to shopify
+    def default_backoff_handler(max_tries: Optional[int], factor: float, **kwargs):
+        logger = AirbyteLogger()
+        def log_retry_attempt(details):
+            _, exc, _ = sys.exc_info()
+            if exc.response:
+                logger.info(f"Status code: {exc.response.status_code}, Response Content: {exc.response.content}")
+            logger.info(
+                f"Caught retryable error '{str(exc)}' after {details['tries']} tries. Waiting {details['wait']} seconds then retrying..."
+            )
+
+        def should_give_up(exc):
+            # If a non-rate-limiting related 4XX error makes it this far, it means it was unexpected and probably consistent, so we shouldn't back off
+            give_up = exc.response is not None and exc.response.status_code != requests.codes.too_many_requests and 400 <= exc.response.status_code < 500
+            if give_up:
+                logger.info(f"Giving up for returned HTTP status: {exc.response.status_code}")
+            elif "CertificateError" in str(exc):
```

Comment:
Is Shopify sending a `CertificateError` with not an error code?  The `if` statement may be first if it uses an error code.

Reply:
We are getting `exc.response` as `None` in case of CertificateError.
exc value in case of CertificateError is:
```
HTTPSConnectionPool(host='rudderstack-store-final.myshopify.com.myshopify.com', port=443): Max retries exceeded with url: /admin/api/2022-10/shop.json?limit=250&order=updated_at+asc&updated_at_min=2023-01-01 (Caused by SSLError(CertificateError("hostname 'rudderstack-store-final.myshopify.com.myshopify.com' doesn't match either of '*.myshopify.com', 'myshopify.com'")))
{"type": "LOG", "log": {"level": "INFO", "message": "Giving up due to certificate error. Please check your shop name."}}
```
Hence I am handling it this way